sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-evaluation-result-processor/cas-active-supplier-not-found-processor.ts

```diff
@@ -0,0 +1,118 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { SystemUsersService } from "@sims/services";
+import { CASSupplier, SupplierStatus } from "@sims/sims-db";
+import { ProcessSummary } from "@sims/utilities/logger";
+import {
+  CASEvaluationResult,
+  CASEvaluationStatus,
+} from "../cas-supplier.models";
+import { Repository } from "typeorm";
+import {
+  CASAuthDetails,
+  CASService,
+  CreateSupplierAndSiteResponse,
+} from "@sims/integrations/cas";
+import { CASEvaluationResultProcessor, ProcessorResult } from ".";
+
+/**
+ * Process a student that was not found on CAS.
+ */
+@Injectable()
+export class CASActiveSupplierNotFoundProcessor extends CASEvaluationResultProcessor {
+  constructor(
+    private readonly casService: CASService,
+    private readonly systemUsersService: SystemUsersService,
+    @InjectRepository(CASSupplier)
+    private readonly casSupplierRepo: Repository<CASSupplier>,
+  ) {
+    super();
+  }
+
+  /**
+   * Create the new supplier and site on CAS using the student information.
+   * @param casSupplier student supplier information from SIMS.
+   * @param evaluationResult evaluation result to be processed.
+   * @param auth authentication token needed for possible
+   * CAS API interactions.
+   * @param summary current process log.
+   * @returns processor result.
+   */
+  async process(
+    casSupplier: CASSupplier,
```

Comment:
Can we just have the supplier id here? So that the processor(s) which needs more details than id can query and get it.

Reply:
I am not sure if I am following. The `casSupplier.id` is present and loaded.
`CASSupplier` as it is right is the full `CASSupplier` object returned from the "get pending suppliers query".
No specific interface was created for this one. Unless some fresh data is needed, all data required to execute the processes should be loaded in advance using the "get pending suppliers query".

Reply:
yes the id is already present in `CASSupplier` object. My point was the param to be just id not the whole object. 

I see 2 processors using only id and one processor using the whole object. So for the processor(s) which need more details can do a get query to find the object by  id.